worker/src/digest_stream.rs

```diff
@@ -0,0 +1,66 @@
+use wasm_bindgen::{self, prelude::*};
+use wasm_bindgen_futures;
+use web_sys::js_sys::{ArrayBuffer, Uint8Array};
+use web_sys::WritableStream;
+
+impl DigestStream {
+    #[allow(dead_code)]
+    pub fn new(algo: DigestStreamAlgorithm) -> Self {
+        Self::internal_new(algo.as_str())
+    }
+
+    #[allow(dead_code)]
+    pub async fn digest(&self) -> Result<Uint8Array, JsValue> {
```

Comment:
I think this should return `Result<Vec<u8>, worker::Error>`. If you need access to the JavaScript value, then maybe add `digest_raw`?

Reply:
d'oh, yup, and this is supported automatically with `?` :)